crates/router/src/core/payments.rs

```diff
@@ -1199,13 +1197,40 @@ pub async fn list_payments(
                     storage_enums::MerchantStorageScheme::PostgresOnly,
                 )
                 .await
-                .ok()?;
-            Some((pi, pa))
-        })
-        .collect::<Vec<(storage::PaymentIntent, storage::PaymentAttempt)>>()
-        .await;
+            {
+                Ok(pa) => Some(Ok((pi, pa))),
+                Err(e) => {
+                    if e.current_context().is_db_not_found() {
+                        logger::warn!(
+                            "payment_attempts missing for payment_id : {} | error : {}",
+                            pi.payment_id,
+                            e
+                        );
+                        return None;
+                    }
+                    Some(Err(e))
+                }
+            }
+        }
+    });
 
-    let data: Vec<api::PaymentsResponse> = pi.into_iter().map(ForeignFrom::foreign_from).collect();
+    //If any of the response are Err, we will get Result<Err(_)>
+    let pi_pa_tuple_vec: Result<Vec<(storage::PaymentIntent, storage::PaymentAttempt)>, _> =
+        join_all(collected_futures)
+            .await
+            .into_iter()
+            .flatten() //Will ignore `None`, will only flatten 1 level
+            .collect::<Result<Vec<(storage::PaymentIntent, storage::PaymentAttempt)>, _>>();
+    //Will collect responses in same order async, leading to sorted responses
+
+    //Converting Intent-Attempt array to Response if no error
+    let data: Vec<api::PaymentsResponse> = match pi_pa_tuple_vec {
+        Ok(vector) => vector.into_iter().map(ForeignFrom::foreign_from).collect(),
+        Err(e) => {
```

Comment:
We are pulling out the error here now, I think putting it as internal server is right here.

Reply:
Hey,
we could use change_context here...

e.g 
```
let data: Vec<api::PaymentsResponse>  = pi_pa_tuple_vec
    .change_context(errors::ApiErrorResponse::InternalServerError)?
    .into_iter()
    .map(ForeignFrom::foreign_from)
    .collect()
```